packages/relay/tests/lib/eth.spec.ts

```diff
@@ -536,6 +536,25 @@ describe('Eth calls using MirrorNode', async function () {
     }
   });
 
+  it('"eth_blockNumber" return the latest block number on second try', async function () {
+    mock.onGet('blocks?limit=1&order=desc').reply(404, {
+      '_status': {
+        'messages': [
+          {
+            'message': 'Block not found'
+          }
+        ]
+      }
+    });
+
+    mock.onGet('blocks?limit=1&order=desc').reply(200, {
```

Comment:
Мисля, че тези можеш да ги чейнеш по този начин `mock.onGet('blocks?limit=1&order=desc').replyOnce(...).onGet('blocks?limit=1&order=desc').replyOnce(...)`

Reply:
Yep, that can works as well. Thanks.

Reply:
Another thing, I believe the way the mocks are currently set, the second one overwrites the first one. If you switch them around the test will fail, even though the first mock would return 200. That's why you have to use `replyOnce`

Reply:
Also currently it does not retry at all, because here https://github.com/hashgraph/hedera-json-rpc-relay/blob/b81fe859f329ac7e97a4122833a487a8e9a9b730/packages/relay/tests/lib/eth.spec.ts#L89 an axios instance is created which does not have the retry option. This isntance is passed to the MirrorNodeClient and the Axios mock, so the MirrorNodeClient. We should not create and pass an axios client instance to both so the MirrorNodeClient will create it's own which includes the retry logic. And then the mock instance should be initialised like this `mock = new MockAdapter(mirrorNodeInstance.client, { onNoMatch: "throwException" });`

Reply:
Done. Now it's using the proper instance.